sui/src/rest_gateway.rs

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use move_core_types::identifier::Identifier;
+use move_core_types::language_storage::TypeTag;
+use serde::de::DeserializeOwned;
+use serde_json::{json, Value};
+
+use sui_core::gateway_state::gateway_responses::TransactionResponse;
+use sui_core::gateway_state::GatewayAPI;
+use sui_types::base_types::{encode_bytes_hex, ObjectID, ObjectRef, SuiAddress};
+use sui_types::messages::{Transaction, TransactionData};
+use sui_types::object::ObjectRead;
+
+use crate::rest_gateway::requests::{
+    CallRequest, MergeCoinRequest, PublishRequest, SplitCoinRequest,
+};
+use crate::rest_gateway::responses::{NamedObjectRef, ObjectResponse, TransactionBytes};
+use async_trait::async_trait;
+pub mod requests;
+pub mod responses;
+
+pub struct RestGatewayClient {
+    pub url: String,
+}
+#[async_trait]
+#[allow(unused_variables)]
+impl GatewayAPI for RestGatewayClient {
+    async fn execute_transaction(
+        &mut self,
+        tx: Transaction,
+    ) -> Result<TransactionResponse, anyhow::Error> {
+        let url = format!("{}/api/execute_transaction", self.url);
+        let data = tx.data.to_base64();
+        let sig_and_pub_key = format!("{:?}", tx.tx_signature);
```

Comment:
Are we planning to provide util functions for the users to construct a signature using private keys?

Reply:
user should be able to construct the signature using provided BCS encoded bytes (from `new_transfer`, `move_call` endpoints etc) with any crypto lib and their ed25519 key.

Reply:
Yes, but as an application developer I don't really know how to do this. It would be great if this is exposed as a library function similar to https://docs.ethers.io/v5/api/signer/#Signer--signing-methods